src/sinks/elasticsearch.rs

```diff
@@ -483,6 +478,24 @@ fn maybe_set_id(key: Option<impl AsRef<str>>, doc: &mut serde_json::Value, event
     }
 }
 
+fn get_and_strip_basic_auth(endpoint: &str) -> Option<(String, String)> {
```

Comment:
I'm not sure this is an improvement over returning `Option<(String, ElasticSearchAuth)>` and restore the apply functionality above.

Reply:
I agree, but `ElasticSearchAuth::apply` was not being used, so I went with a minimal change instead.

Reply:
I moved `get_and_strip_basic_auth` to `src/http.rs`, it now returns `Auth`.